sympy/ntheory/generate.py

```diff
@@ -246,9 +263,32 @@ def primepi(n):
     n = int(n)
     if n < 2:
         return 0
-    else:
-        return sieve.search(n)[0]
-
+    lim = int(n ** 0.5)
+    lim -= 1
+    lim = max(lim,0)
+    while lim * lim <= n:
+        lim+=1
+    lim-=1
+    arr1 = [0]*(lim+1)
+    arr2 = [0]*(lim+1)
+    for i in range(1, lim+1):
+        arr1[i] = i-1
```

Comment:
Perhaps there could be a bit longer explanation of ``phi(i, j)`` since it does not appear to be the same as e.g. in the wikipedia article. Especially, it seems that not all numbers ``<= i`` are included in this first step.